src/CoreWCF.MSMQ/src/CoreWCF/Channels/MsmqBindingElementBase.cs

```diff
@@ -0,0 +1,199 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using CoreWCF.Queue.Common.Configuration;
+using CoreWCF.Runtime;
+
+namespace CoreWCF.Channels
+{
+    public abstract class MsmqBindingElementBase : QueueBaseTransportBindingElement
+    {
+        private DeadLetterQueue _deadLetterQueue;
+        private int _maxRetryCycles;
+        private int _receiveRetryCount;
+        private TimeSpan _retryCycleDelay;
+        private TimeSpan _timeToLive;
+
+        //private ReceiveErrorHandling receiveErrorHandling;
+
+        public Uri CustomDeadLetterQueue { get; set; }
+
+        public DeadLetterQueue DeadLetterQueue
+        {
+            get { return _deadLetterQueue; }
+            set
+            {
+                if (!DeadLetterQueueHelper.IsDefined(value))
+                {
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentOutOfRangeException("value"));
+                }
+
+                _deadLetterQueue = value;
+            }
+        }
+
+        public bool Durable { get; set; }
+
+        public bool TransactedReceiveEnabled => ExactlyOnce;
+
```

Comment:
This is an implementation of ITransactedReceiveBindingElement which we don't have yet so you can remove this property.